core/src/main/java/ai/timefold/solver/core/impl/domain/lookup/LookUpStrategyResolver.java

```diff
@@ -81,17 +38,17 @@ public LookUpStrategyResolver(DescriptorPolicy descriptorPolicy, LookUpStrategyT
      */
     public LookUpStrategy determineLookUpStrategy(Object object) {
         return decisionCache.computeIfAbsent(object.getClass(), objectClass -> {
-            if (objectClass.isEnum()) {
+            if (DeepCloningUtils.isImmutable(objectClass)) {
```

Comment:
I'm wondering... should we check for arrays here as well? Those can't possible be annotated.

Reply:
I  think value ranges that return arrays are rare, but probably not impossible. Arrays are mutable though, and may contain reference to mutable values (i.e. PlanningEntities).

Reply:
That's true, but do they need to be looked up?
(The array check wouldn't be inside of `isImmutable`.)

Reply:
Yes, they will be looked up in `ChangeMove.rebase/SwapMove.rebase`.